Junk4Dollars-app/Auctions/AuctionListTableViewDataSource.swift

```diff
@@ -10,11 +10,17 @@ public class AuctionListTableViewDataSource: NSObject, UITableViewDataSource {
 
     public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = UITableViewCell()
-        cell.textLabel?.text = auctions[indexPath.row].title
+        cell.textLabel?.text = auctionFor(indexPath: indexPath).title
         return cell
     }
     
     public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return auctions.count
     }
+
+    public func auctionFor(indexPath: IndexPath, callback:( (_ id: Int) -> Void)? = nil) -> Auction {
+        let auction = auctions[indexPath.row]
+        callback?(auction.identifier)
```

Comment:
Why is it that you need the callback? The method is already returning the Auction, so callers will have access to the id anyways.

Reply:
When a cell is clicked, the delegate method `tableView didSelectRowAt` runs, and that method returns void. I'm having that method call this method inside the data source so that it can figure out which auction was in the tapped cell. That auction id will need to be passed to the next view that renders a single auction. I wasn't sure how else to get that `id` out of here